lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,523 @@
+        if not lastName:
+            self.logger.error(f"Register User: missing lastName field")
+            abort(
+                400, message=f"missing lastName field",
+            )
+
+        # validate the email field
+        try:
+            valid = validate_email(emailID)
+            email = valid.email
+        except EmailNotValidError as e:
+            self.logger.error(f"Register User: invalid email {emailID}")
+            abort(400, message=f"Invalid email: {emailID}")
+
+        # check if user already exist
+        user = UserModel.query.filter_by(username=user_data.get("username")).first()
+        if not user:
+            try:
+                user = UserModel(
+                    bcrypt=self.bcrypt,
+                    bcrypt_log_rounds=self.bcrypt_log_rounds,
+                    username=username.lower(),
+                    password=password,
+                    firstName=firstName,
+                    lastName=lastName,
+                    email=email,
+                )
+
+                # insert the user
+                self.db_session.add(user)
+                self.db_session.commit()
+                self.logger.info(
+                    f"New user registered, Name: {firstName} {lastName} username: {username}, email: {email}"
+                )
+
+                # generate the auth token
+                auth_token = user.encode_auth_token(
+                    self.token_expire_duration, user.id, self.logger
+                )
+                if not auth_token:
+                    self.logger.info(
+                        f"User registered but unable to encode the auth token. username {username}"
+                    )
+                    return make_response(
+                        "{'message':'INTERNAL ERROR, unable to encode the auth token, please try logging in}",
+                        500,
+                    )
+
+                self.logger.info(auth_token.decode())
+                response_object = {
+                    "status": "success",
+                    "message": "Successfully registered.",
+                    "auth_token": auth_token.decode(),
+                }
+                response = jsonify(response_object)
+                response.status_code = 201
+                return make_response(response, 201)
+                # return Response(response_object, status=201, mimetype="application/json")
+
+            except (
+                SQLAlchemyError,
+                DataError,
+                DatabaseError,
+                DBAPIError,
+                CompileError,
+            ) as e:
+                self.logger.error(
+                    f"Some SQLAlchemy error occurred while registering a user {e.__dict__['orig']}"
+                )
+                abort(500, message="INTERNAL ERROR, Please try again.")
+
+            except Exception as e:
+                self.logger.error(f"Exception while registering a user {e}")
```

Comment:
Let's use `self.logger.exception("Exception while registering a user")` instead of trying to minimize the error.  Internal errors arguably should product stack traces on the server side so we can figure out what went wrong.

Also, we don't want to use f-strings with logging statements as a rule so that we don't spend time formatting strings before they are emitted.  In this case, that may seem silly to apply, because we will likely always emit such an error message.  The issue comes when we lose the discipline of maintaining that practice as we create "debug" or "info" messages which are huge but not logged (have had many performance problems with python applications that took a while to debug end up being string formatting for logging statements that are not emitted).